rollout/controller.go

```diff
@@ -197,6 +213,12 @@ func (c *RolloutController) Run(threadiness int, stopCh <-chan struct{}) error {
 		}, time.Second, stopCh)
 	}
 	log.Info("Started Rollout workers")
+
+	gvk := schema.ParseGroupResource("virtualservices.networking.istio.io").WithVersion(c.defaultIstioVersion)
+	go wait.Until(func() {
+		controllerutil.WatchResource(c.dynamicclientset, gvk, c.rolloutWorkqueue, c.rolloutsIndexer, virtualServiceIndexName)
+	}, time.Minute, stopCh)
```

Comment:
Related to my other comment. The use of `wait.Until` with an indiscriminate 1 minute retry is too simplistic. I would switch this to use exponential backoff, which will work well in all circumstances. (e.g. if we discover the resource isn't in the cluster, then we could eventually backoff to only check every 5 min, reducing API call rate).